src/module.rs

```diff
@@ -289,7 +367,17 @@ impl Module {
         MemoryBuffer::new(memory_buffer)
     }
 
+    /// Ensures that the current `Module` is valid, and returns a `bool`
+    /// that describes whether or not it is.
+    /// 
+    /// * `print` - Whether or not a message describing the error should be printed
+    ///   to stderr, in case of failure.
+    /// 
+    /// # Remarks
+    /// See also: http://llvm.org/doxygen/Analysis_2Analysis_8cpp_source.html
     pub fn verify(&self, print: bool) -> bool {
+        // REVIEW <6A>: Maybe, instead of printing the module to stderr with `print`,
+        // we should return Result<(), String> that contains an error string?
```

Comment:
Yeah I think There's another variant `LLVMVerifierFailureAction::LLVMAbortProcessAction` too, which prints to stderr and aborts the process. Not sure if we want to support that. But `Result<(), String>` does seem reasonable.

I also haven't been able to decide if we should return `&CStr`s because they point to already allocated memory or if we should reallocate to a `String` and dispose the original. A bigger design question for another time I suppose.

Reply:
I've been thinking about `&CStr`'s as well; in any case the user probably won't use them without `as_str` or `to_string_lossy` for example, and since `&str` is used everywhere else, it is rather strange; but it has indeed a performance cost.

In any case, I think aborting should be left to the user.